boa_engine/src/symbol.rs

```diff
@@ -34,7 +34,7 @@ use num_enum::{IntoPrimitive, TryFromPrimitive};
 use std::{
     hash::{Hash, Hasher},
     sync::{
-        atomic::{AtomicU64, Ordering},
+        atomic::{AtomicUsize, Ordering},
```

Comment:
The problem with `usize` is that on 32-bit architectures this is a `u32`, which isn't big enough, we can reach the max in about `15s`. With `u64` will take more than a century.

This effects the whole program, so all threads can't create `JsSymbol`s

Reply:
So we use `portable_atomic::AtomicU64`? I considered that alternative, but opted for this because `AtomicU64` is a lot slower on platforms without 8 byte atomics, since they have to get a global lock to do operations.

Reply:
I think the question would be: do we want to prioritise flexibility or speed?

Reply:
Thinking about it more thoroughly, I don't think we need to pick one of them, we can just leave the decision to our users!

A "fast-symbols" feature that selects `AtomicUsize` if it is enabled and fallbacks to `AtomicU64` otherwise should be enough. What do you think?

Reply:
You could also argue that symbols are not as prevalent in most JS programs, so it would be pretty rare to have `u32::MAX` symbols alive at the same time.

Reply:
> I think the question would be: do we want to prioritise flexibility of use or speed?

I'm not 100% into the topic at the moment, but I'd guess 8 byte atomics are common on most platforms nowdays? I would say we go for the flexibility right now as  the speed cost would not affect most users. If we notice it to be a problem we could always add optimizations or even give the users the option to choose, right?

Reply:
> > I think the question would be: do we want to prioritise flexibility of use or speed?
> 
> I'm not 100% into the topic at the moment, but I'd guess 8 byte atomics are common on most platforms nowdays? I would say we go for the flexibility right now as  the speed cost would not affect most users. If we notice it to be a problem we could always add optimizations or even give the users the option to choose, right?

Ok, since it seems like both of you prefer the `AtomicU64` option, I'll prioritise flexibility then. I think it shouldn't affect performance for most programs because most of them don't create a lot of symbols consecutively.

Reply:
Yes, lets continue with `AtomicU64`, if we need to we can change it in the future :)